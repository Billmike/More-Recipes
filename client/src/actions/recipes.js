import axios from 'axios';
import {
  ADD_RECIPE,
  GET_RECIPES,
  GET_ONE_RECIPE,
  GET_USER_RECIPES,
  ADD_FAVORITE_RECIPE,
  FETCH_FAVORITE_RECIPES,
  EDIT_RECIPE,
  REMOVE_RECIPE,
  ADD_REVIEW,
  UPVOTE_RECIPE,
  DOWNVOTE_RECIPE
} from './types';
import './toastrConfig';

export const addRecipe = recipe => ({
  type: ADD_RECIPE,
  recipe
});

export const getAllRecipes = (recipes) => {
  return {
    type: GET_RECIPES,
    recipes,
  };
};

export const getOneRecipe = (recipe) => {
  return {
    type: GET_ONE_RECIPE,
    id: recipe.id,
    recipe
  };
};

export const getUserRecipe = (userRecipe) => {
  return {
    type: GET_USER_RECIPES,
    userRecipe,
  };
};

export const addFavorites = (favoriteRecipes) => {
  return {
    type: ADD_FAVORITE_RECIPE,
    favoriteRecipes
  };
};

export const fetchFavorites = (favoriteRecipes) => {
  return {
    type: FETCH_FAVORITE_RECIPES,
    favoriteRecipes,
  };
};

export const editRecipe = (id, updates) => ({
  type: EDIT_RECIPE,
  id,
  updates,
});

export const removeRecipe = ({ id } = {}) => ({
  type: REMOVE_RECIPE,
  id
});

export const addReview = (review) => {
  return {
    type: ADD_REVIEW,
    review,
  };
};

export const upVoteRecipe = (id) => {
  return {
    type: UPVOTE_RECIPE,
    id,
  };
};

export const downVoteRecipe = (id) => {
  return {
    type: DOWNVOTE_RECIPE,
    id,
  };
};

export const startGetAllRecipes = () => {
  return (dispatch) => {
    axios.get('http://localhost:8000/api/v1/recipes')
      .then((res) => {
        dispatch(getAllRecipes(res.data.recipeData));
      })
      .catch((error) => {
        console.log(error);
      });
  };
}

export const startAddRecipe = (recipeData = {}) => {
  return (dispatch) => {
    const {
      name = '',
      description = '',
      img_url = 'no-img-here',
      category = '',
      ingredients = '',
      instructions = ''
    } = recipeData;

    const recipe = {
      name,
      description,
      img_url,
      category,
      ingredients,
      instructions,
    };
    axios.post('http://localhost:8000/api/v1/recipes', recipe)
      .then(() => {
        toastr.success('Recipe added successfully.');
        dispatch(addRecipe({
          id: recipe.id,
          ...recipe
        }));
      });
  };
};

export const startGetUserRecipes = () => {
  return (dispatch) => {
    axios.get('http://localhost:8000/api/v1/users/recipes')
      .then((res) => {
        dispatch(getUserRecipe(res.data.userRecipe));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export const startEditRecipe = (id, updates) => {
  return (dispatch) => {
    axios.put(`http://localhost:8000/api/v1/recipes/${id}/modify`, updates)
      .then(() => {
        toastr.success('Recipe edited successfully.');
        dispatch(editRecipe(id, updates));
      })
      .catch((error) => {
        console.log('_<_<_<_<_<__<_', error);
      });
  };
};

export const startRemoveRecipe = (id) => {
  return (dispatch) => {
    axios.delete(`http://localhost:8000/api/v1/recipes/${id}`)
      .then(() => {
        toastr.success('Recipe deleted successfully.');
        dispatch(removeRecipe(id));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export const startGetOneRecipe = (id) => {
  return (dispatch) => {
    axios.get(`http://localhost:8000/api/v1/recipes/${id}`)
      .then((res) => {
        dispatch(getOneRecipe(res.data.recipeData));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export const startAddFavoriteRecipes = (id) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/favorites`)
      .then((res) => {
        toastr.success(res.data.message);
        return dispatch(addFavorites(res.data));
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'Sorry, you cannot perform this action on your own recipe.') {
          toastr.error(err.response.data.message);
        }
      });
  };
};

export const startGetUserFavorites = (id) => {
  return (dispatch) => {
    axios.get(`http://localhost:8000/api/v1/users/${id}/favorites`)
      .then((res) => {
        dispatch(fetchFavorites(res.data));
      })
      .catch((err) => {
        console.log(err);
      });
  };
};

export const startUpvoteRecipe = (id) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/votes/upvote`)
      .then((res) => {
        toastr.success(res.data.message);
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'You already upvoted on this recipe') {
          toastr.warning(err.response.data.message);
        }
      });
  };
};

export const startDownVoteRecipe = (id) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/votes/downvote`)
      .then((res) => {
        toastr.success(res.data.message);
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'You already downvoted on this recipe') {
          toastr.warning(err.response.data.message);
        }
      });
  };
};

export const startAddReview = (id, data) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/reviews`, data)
      .then((res) => {
        dispatch(addReview(res.data));
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'You can\'t post an empty review. Please, enter a happy review for this recipe.') {
          toastr.warning(err.response.data.message);
        }
      });
  };
};
