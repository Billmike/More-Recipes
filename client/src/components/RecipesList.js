import React from 'react';
import pizza from '../assets/img/pizzza.jpg';

const RecipeList = ({ name, description, category }) => {
	return (
		<div className="card-group">
        <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src={ pizza } alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">{name}</h4></a>
              <p className="card-text">{description}</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></a>
              <a href="#" className="btn border border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  89 <i className="fa fa-heart" aria-hidden="true" /></a>
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Recipe Category - {category}</small>
            </div>
          </div>
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src={ pizza } alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">{name}</h4></a>
              <p className="card-text">{description}</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></a>
              <a href="#" className="btn border border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  89 <i className="fa fa-heart" aria-hidden="true" /></a>
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Recipe Category - {category}</small>
            </div>
          </div>
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src={ pizza } alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">{name}</h4></a>
              <p className="card-text">{description}</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></a>
              <a href="#" className="btn border border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  89 <i className="fa fa-heart" aria-hidden="true" /></a>
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Recipe Category - {category}</small>
            </div>
          </div>
      </div>
		)
};

export default RecipeList;
