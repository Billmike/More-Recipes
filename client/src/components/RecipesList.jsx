import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const RecipeList = ({ id, name, description, category }) => (
  <div className="card-group">
      <div className="card index-card">
        <Link to={`/recipe/${id}`}><img className="card-img-top" alt="Pizza" /></Link>
        <div className="card-body">
          <Link to={`/recipe/${id}`}><h4 className="card-title">{name}</h4></Link>
          <p className="card-text">{description}</p>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">  89 <i className="fa fa-heart" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></Link>
        </div>
        <div className="card-footer">
          <small className="text-muted">Recipe Category - {category}</small>
        </div>
      </div>
      <div className="card index-card">
        <Link to={`/recipe/${id}`}><img className="card-img-top" alt="Pizza" /></Link>
        <div className="card-body">
          <Link to={`/recipe/${id}`}><h4 className="card-title">{name}</h4></Link>
          <p className="card-text">{description}</p>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">  89 <i className="fa fa-heart" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></Link>
        </div>
        <div className="card-footer">
          <small className="text-muted">Recipe Category - {category}</small>
        </div>
      </div>
      <div className="card index-card">
        <Link to={`/recipe/${id}`}><img className="card-img-top" alt="Pizza" /></Link>
        <div className="card-body">
          <Link to={`/recipe/${id}`}><h4 className="card-title">{name}</h4></Link>
          <p className="card-text">{description}</p>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">  89 <i className="fa fa-heart" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></Link>
        </div>
        <div className="card-footer">
          <small className="text-muted">Recipe Category - {category}</small>
        </div>
      </div>
  </div>
);

export default connect()(RecipeList);
