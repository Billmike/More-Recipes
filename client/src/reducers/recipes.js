const recipeDefaultState = [];

export default (state = recipeDefaultState, action) => {
  switch (action.type) {
    case 'ADD_RECIPE':
      return [
        ...state,
        action.recipe,
      ];
    default:
      return state;
  }
};
