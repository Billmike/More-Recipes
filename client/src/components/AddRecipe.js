import React from 'react';
import RecipesForm from './RecipesForm';
import { connect } from 'react-redux';
import { addRecipe } from '../actions/recipes';

const AddRecipe = (props) => {
    console.log(props)
	return (
	   <div>
       <h1 className="container">Add Recipe</h1>
        <RecipesForm onSubmit={(recipe) => {
            props.dispatch(addRecipe(recipe));
            props.history.push('/')
        }} />
       </div>	
	)
}

export default connect()(AddRecipe);
