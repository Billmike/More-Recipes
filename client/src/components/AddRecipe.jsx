import React, { Component } from 'react';
import RecipesForm from './RecipesForm';
import { connect } from 'react-redux';
import { addRecipe } from '../actions/recipes';

export class AddRecipe extends Component {
  onSubmit = (recipe) => {
    this.props.addRecipe(recipe);
    this.props.history.push('/dashboard');
  };

  render() {
    return (
      <div>
        <h1 className="container"> Add Recipe </h1>
        <RecipesForm
          onSubmit={this.onSubmit}
        />
      </div>
      );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    addRecipe: (recipe) => dispatch(addRecipe(recipe))
  }
}

// const AddRecipe = (props) => {
//     console.log(props)
// 	return (
// 	   <div>
//        <h1 className="container">Add Recipe</h1>
//         <RecipesForm onSubmit={(recipe) => {
//             props.dispatch(addRecipe(recipe));
//             props.history.push('/')
//         }} />
//        </div>	
// 	)
// }

export default connect(undefined, mapDispatchToProps)(AddRecipe);
