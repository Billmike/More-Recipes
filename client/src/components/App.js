import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Homepage from './Hompage';
import Dashboard from './Dashboard';
import Favorite from './Favorite';
import AddRecipe from './AddRecipe';
import EditRecipe from './EditRecipe';
import ProfilePage from './ProfilePage';
import NotFoundPage from './NotFoundPage';

class App extends Component {
    render() {
        return(
            <div>
                <BrowserRouter>
                    <div>
                        <Navbar />
                        <Switch>
                        <Route exact path="/" component={Homepage} />
                        <Route path="/dashboard" component={Dashboard} />
                        <Route path="/favorite" component={Favorite} />
                        <Route path="/add" component={AddRecipe} />
                        <Route path="/edit" component={EditRecipe} />
                        <Route path="/profile" component={ProfilePage} />
                        <Route component={NotFoundPage} />
                    </Switch>
                    </div>
                </BrowserRouter>
            </div>
        )
    }
}

export default App;
