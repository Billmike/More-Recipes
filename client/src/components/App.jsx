import React from 'react';
import { BrowserRouter, browserHistory, Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Homepage from './Hompage';
import Dashboard from './Dashboard';
import RecipeDetails from './RecipeDetails';
import Favorite from './Favorites';
import AddRecipe from './AddRecipe';
import EditRecipe from './EditRecipe';
import ProfilePage from './ProfilePage';
import LoginPage from './LoginPage';
import RegistrationPage from './RegistrationPage';
import RequireAuth from '../utils/RequireAuth';
import NotFoundPage from './NotFoundPage';

const App = () => (
  <div>
    <BrowserRouter history={browserHistory}>
      <div>
        <Navbar />
        <Switch>
          <Route exact path="/" component={Homepage} />
          <Route path="/dashboard" component={RequireAuth(Dashboard)} />
          <Route path="/favorites" component={RequireAuth(Favorite)} />
          <Route path="/add" component={RequireAuth(AddRecipe)} />
          <Route path="/edit/:id" component={RequireAuth(EditRecipe)} />
          <Route path="/recipe/:id" component={RecipeDetails} />
          <Route path="/profile" component={RequireAuth(ProfilePage)} />
          <Route path="/login" component={LoginPage} />
          <Route path="/register" component={RegistrationPage} />
          <Route component={NotFoundPage} />
        </Switch>
      </div>
    </BrowserRouter>
  </div>
);

export default App;
