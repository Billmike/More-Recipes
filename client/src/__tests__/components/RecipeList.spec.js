import React from 'react';
import { shallow } from 'enzyme';
import { RecipeList } from '../../components/RecipesList';
import recipes from '../fixtures/recipes';

describe('<RecipesList />', () => {
  it('Should mount the Recipe List component correctly', () => {
    const wrapper = shallow(<RecipeList recipe={recipes[2]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
