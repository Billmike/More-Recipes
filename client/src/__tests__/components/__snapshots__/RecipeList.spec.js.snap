// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipesList /> Should mount the Recipe List component correctly 1`] = `
<div
  className="col-md-4"
>
  <div
    className="card"
  >
    <Link
      replace={false}
      to="/recipe/3"
    >
      <img
        alt="Pizza"
        className="card-img-top"
      />
    </Link>
    <div
      className="card-body"
    >
      <Link
        replace={false}
        to="/recipe/3"
      >
        <h4
          className="card-title"
        >
          Amazing fried rice
        </h4>
      </Link>
      <p
        className="card-text"
      >
        A really amazing fried rice
      </p>
      <div
        className="group-btn"
      >
        <Link
          className="btn border border-secondary rounded"
          replace={false}
          to="/recipe/3"
        >
          <i
            aria-hidden="true"
            className="fa fa-eye"
          />
        </Link>
        <Link
          className="btn border border-secondary rounded"
          onClick={[Function]}
          replace={false}
          to="/recipe/3"
        >
           
          1
          <i
            aria-hidden="true"
            className="fa fa-heart"
          />
        </Link>
        <Link
          className="btn border border-secondary rounded"
          replace={false}
          to="/recipe/3"
        >
           
          1
           
          <i
            aria-hidden="true"
            className="fa fa-comments"
          />
        </Link>
      </div>
    </div>
    <div
      className="card-footer pad-footer"
    >
      <small
        className="text-muted small-text category-text"
      >
        Recipe Category - 
        Lunch
      </small>
    </div>
  </div>
</div>
`;
